### PR TITLE
1644 fix geocoder issues

### PR DESCRIPTION
**Pull request by briaguya**
Ended up doing quite a bit of refactoring to get this working to the point I have. Haven't tested everything but it seems to address most of the issues in https://github.com/cityofaustin/techstack/issues/1644

### CODE REVIEW

**Comment by easherma** [comment]
Make some good ground with some refactoring, cleaning up imports. 
This helped me set up some geocoding debugging that I wished we could do earlier, on our way to being able to switch between geocoding engines as needed.

File: src/js/widgets/SelectLocation/SelectLocationWidget.jsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 import ReactMapboxGl, { Layer } from 'react-mapbox-gl';
-import * as MapboxGl from 'mapbox-gl';
+import { NavigationControl } from 'mapbox-gl';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
 
-// importing the geocoder didnt seem to work at first
-const MapboxGeocoder = require('@mapbox/mapbox-gl-geocoder');
+const HERE_APP_ID = 'R3EtGwWQmTKG5eVeyLV8';
+const HERE_APP_CODE = '8aDkNeOzfxGFkOKm9fER0A';
+const MAPBOX_TOKEN = 'pk.eyJ1IjoiY3Jvd2VhdHgiLCJhIjoiY2o1NDFvYmxkMHhkcDMycDF2a3pseDFpZiJ9.UcnizcFDleMpv5Vbv8Rngw';
```
Comment (Niicck) [inline src/js/widgets/SelectLocation/SelectLocationWidget.jsx:8]: Is it safe to have these tokens saved in a public repo?

File: src/js/widgets/SelectLocation/SelectLocationWidget.jsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+const GEOCODE_DEBUG = false;
```
Comment (Niicck) [inline src/js/widgets/SelectLocation/SelectLocationWidget.jsx:9]: I'm getting a ton of console logs even with GEOCODER_DEBUG set to false.

File: src/js/widgets/SelectLocation/SelectLocationWidget.jsx (hunk 1 above)
Comment (easherma) [reply src/js/widgets/SelectLocation/SelectLocationWidget.jsx:8]: They are exposed client side, so there is little choice. However, I would eventually like to import them from a .env file or some other config method to enable better resusability by others.

File: src/js/widgets/SelectLocation/SelectLocationWidget.jsx (hunk 2 above)
Comment (easherma) [reply src/js/widgets/SelectLocation/SelectLocationWidget.jsx:9]: Yah, if the form was still built with the 'true' flag, that would happen.

**Pull request merged**